### ISSUE TITLE
[Feat request] theme files support

### ISSUE THREAD

**Issue by gusano** [supercollider/supercollider#3411]
It would be great to be able to add themes by adding them to `~/.config/SuperCollider/themes/*.yaml` instead of harcoded in `editors/sc-ide/core/settings/theme.cpp`.
We could keep *default* one hardcoded as a fallback.

**Comment by gusano** [comment]
@alln4tural the idea would be to have themes **outside** from `sc_ide_conf.yaml` so it's so much easier to add new ones without messing up the main config file.

**Comment by patrickdupuis** [comment]
Having themes as separate files also means that it is easier for us to include the proper license information and links to the original projects repo/website.

**Comment by snappizz** [comment]
would recommend support for the `.tmTheme` file format. having trouble locating any sort of spec for it though.

**Comment by gusano** [comment]
> would recommend support for the .tmTheme file format

IMHO it would be much easier to read/parse `*.yaml` files

**Comment by brianlheim** [comment]
> > would recommend support for the .tmTheme file format
>
> IMHO it would be much easier to read/parse *.yaml files

I agree. We already have code for serializing the themes to YAML. tmtheme format support can always be added later.

**Comment by brianlheim** [comment]
The theme files would still have to live in the application folder I think. So, at runtime, the IDE would parse YAML files from both the application's resources directory and the `themes` directories of the user and system app directories.

**Comment by brianlheim** [comment]
And for legacy support, read themes from `sc_ide_conf.yaml`. Presumably, when shutting down, themes would be written out to separate files. Here's a sketch of that process:

1. If a theme has the name of a default theme, don't save it out (these are "read-only")
2. Else, if a theme has the name of a theme in the system support directory, overwrite it
3. Else, write it to a file in the user support directory
4. (Write `sc_ide_conf.yaml` as usual, with no themes)

Another option would be to require a theme file to exist before modifying it. This would have the advantage that we always know exactly where to save which theme file, and we can also separate the filename from the name of the theme.
